### ISSUE TITLE
Unable to install Preview.js using VSCode Extensions Marketplace

### ISSUE THREAD

**Issue by TheAussieStew** [fwouts/previewjs#312]
### Describe the bug

While setting up a new computer, and installing the Preview.js extension on VSCode, I get this error:

```
Please wait while Preview.js installs dependencies. This could take a minute.

Dependencies will be installed in: /Users/kongweiying/.vscode/extensions/zenclabs.previewjs-1.4.1/dist/installed

$ npm install


Oh no, it looks like installation failed!

Error: Command failed with ENOENT: npm install
spawn npm ENOENT
```

I have npm installed using brew, and the npm command works fine (although I don't use npm in this particular project, I've installed it because it seems like the extension depends on it).

Looking forward to seeing the new Preview.js extension :) (have been stuck on version 0.7.1 of the old React Preview extension!)

### Reproduction

1. Install the VSCode Preview.js extension
2. Refer to "Output" tab in the VSCode integrated console

### Preview.js version

v1.4.1

### System Info

```shell
System:
    OS: macOS 12.2
    CPU: (8) arm64 Apple M1
    Memory: 76.36 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 17.7.2 - /opt/homebrew/bin/node
    Yarn: 1.22.17 - /opt/homebrew/bin/yarn
    npm: 8.5.2 - /opt/homebrew/bin/npm
  IDEs:
    Nano: 2.0.6 - /usr/bin/nano
    Vim: 8.2 - /usr/bin/vim
    Xcode: /undefined - /usr/bin/xcodebuild
  Browsers:
    Chrome: 99.0.4844.74
    Safari: 15.3
```

### Used Package Manager

yarn

### Extension logs (useful for crashes)

```shell
[2022-03-19 13:44:14.291] [exthost] [info] Extension host with pid 44525 started
[2022-03-19 13:44:14.291] [exthost] [info] Skipping acquiring lock for /Users/kongweiying/Library/Application Support/Code/User/workspaceStorage/1f9645489d518dd927362d967cebcd0f.
[2022-03-19 13:44:14.323] [exthost] [info] ExtensionService#_doActivateExtension vscode.microsoft-authentication, startup: false, activationEvent: 'onAuthenticationRequest:microsoft'
[2022-03-19 13:44:14.330] [exthost] [info] ExtensionService#_doActivateExtension vscode.debug-auto-launch, startup: true, activationEvent: '*'
[2022-03-19 13:44:14.332] [exthost] [info] ExtensionService#_doActivateExtension vscode.git-base, startup: true, activationEvent: '*', root cause: vscode.github
[2022-03-19 13:44:14.333] [exthost] [info] ExtensionService#_doActivateExtension vscode.ipynb, startup: true, activationEvent: '*'
[2022-03-19 13:44:14.335] [exthost] [info] ExtensionService#_doActivateExtension vscodevim.vim, startup: true, activationEvent: '*'
[2022-03-19 13:44:14.355] [exthost] [info] ExtensionService#_doActivateExtension vscode.npm, startup: true, activationEvent: 'workspaceContains:package.json'
[2022-03-19 13:44:14.396] [exthost] [info] ExtensionService#_doActivateExtension vscode.git, startup: true, activationEvent: '*'
[2022-03-19 13:44:14.404] [exthost] [info] ExtensionService#_doActivateExtension vscode.github, startup: true, activationEvent: '*'
[2022-03-19 13:44:14.420] [exthost] [info] ExtensionService#_doActivateExtension vscode.github-authentication, startup: false, activationEvent: 'onAuthenticationRequest:github'
[2022-03-19 13:44:14.471] [exthost] [info] Eager extensions activated
[2022-03-19 13:44:14.480] [exthost] [info] ExtensionService#_doActivateExtension vscode.emmet, startup: false, activationEvent: 'onStartupFinished'
[2022-03-19 13:44:14.497] [exthost] [info] ExtensionService#_doActivateExtension vscode.merge-conflict, startup: false, activationEvent: 'onStartupFinished'
[2022-03-19 13:44:14.499] [exthost] [info] ExtensionService#_doActivateExtension zenclabs.reactpreview, startup: false, activationEvent: 'onStartupFinished'
[2022-03-19 13:45:34.576] [exthost] [info] ExtensionService#_doActivateExtension vscode.typescript-language-features, startup: false, activationEvent: 'onLanguage:typescriptreact'
[2022-03-19 13:46:14.748] [exthost] [error] [zenclabs.reactpreview] provider FAILED
[2022-03-19 13:46:14.748] [exthost] [error] Error: Connection got disposed.
    at Object.dispose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:41363)
    at Object.dispose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:121446)
    at w.handleConnectionClosed (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:121659)
    at w.handleConnectionClosed (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:182005)
    at t (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:119748)
    at r.invoke (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:42976)
    at o.fire (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:43737)
    at Y (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:30621)
    at r.invoke (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:42976)
    at o.fire (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:43737)
    at d.fireClose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:51618)
    at ChildProcess.<anonymous> (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:62579)
    at ChildProcess.emit (events.js:327:22)
    at maybeClose (internal/child_process.js:1048:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
[2022-03-19 13:46:14.748] [exthost] [error] [zenclabs.reactpreview] provider FAILED
[2022-03-19 13:46:14.748] [exthost] [error] Error: Connection got disposed.
    at Object.dispose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:41363)
    at Object.dispose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:121446)
    at w.handleConnectionClosed (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:121659)
    at w.handleConnectionClosed (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:182005)
    at t (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:119748)
    at r.invoke (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:42976)
    at o.fire (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:43737)
    at Y (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:30621)
    at r.invoke (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:42976)
    at o.fire (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:43737)
    at d.fireClose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:51618)
    at ChildProcess.<anonymous> (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:62579)
    at ChildProcess.emit (events.js:327:22)
    at maybeClose (internal/child_process.js:1048:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
[2022-03-19 13:46:14.748] [exthost] [error] [zenclabs.reactpreview] provider FAILED
[2022-03-19 13:46:14.748] [exthost] [error] Error: Connection got disposed.
    at Object.dispose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:41363)
    at Object.dispose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:121446)
    at w.handleConnectionClosed (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:121659)
    at w.handleConnectionClosed (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:182005)
    at t (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:119748)
    at r.invoke (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:42976)
    at o.fire (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:43737)
    at Y (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:30621)
    at r.invoke (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:42976)
    at o.fire (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:43737)
    at d.fireClose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:51618)
    at ChildProcess.<anonymous> (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:62579)
    at ChildProcess.emit (events.js:327:22)
    at maybeClose (internal/child_process.js:1048:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
[2022-03-19 13:48:15.055] [exthost] [error] [zenclabs.reactpreview] provider FAILED
[2022-03-19 13:48:15.055] [exthost] [error] Error: Connection got disposed.
    at Object.dispose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:41363)
    at Object.dispose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:121446)
    at w.handleConnectionClosed (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:121659)
    at w.handleConnectionClosed (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:182005)
    at t (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:119748)
    at r.invoke (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:42976)
    at o.fire (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:43737)
    at Y (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:30621)
    at r.invoke (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:42976)
    at o.fire (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:43737)
    at d.fireClose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:51618)
    at ChildProcess.<anonymous> (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:62579)
    at ChildProcess.emit (events.js:327:22)
    at maybeClose (internal/child_process.js:1048:16)
    at Socket.<anonymous> (internal/child_process.js:439:11)
    at Socket.emit (events.js:315:20)
    at Pipe.<anonymous> (net.js:673:12)
[2022-03-19 13:48:15.055] [exthost] [error] [zenclabs.reactpreview] provider FAILED
[2022-03-19 13:48:15.055] [exthost] [error] Error: Connection got disposed.
    at Object.dispose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:41363)
    at Object.dispose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:121446)
    at w.handleConnectionClosed (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:121659)
    at w.handleConnectionClosed (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:182005)
    at t (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:119748)
    at r.invoke (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:42976)
    at o.fire (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:43737)
    at Y (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:30621)
    at r.invoke (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:42976)
    at o.fire (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:43737)
    at d.fireClose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:51618)
    at ChildProcess.<anonymous> (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:62579)
    at ChildProcess.emit (events.js:327:22)
    at maybeClose (internal/child_process.js:1048:16)
    at Socket.<anonymous> (internal/child_process.js:439:11)
    at Socket.emit (events.js:315:20)
    at Pipe.<anonymous> (net.js:673:12)
[2022-03-19 13:48:15.055] [exthost] [error] [zenclabs.reactpreview] provider FAILED
[2022-03-19 13:48:15.055] [exthost] [error] Error: Connection got disposed.
    at Object.dispose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:41363)
    at Object.dispose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:121446)
    at w.handleConnectionClosed (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:121659)
    at w.handleConnectionClosed (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:182005)
    at t (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:119748)
    at r.invoke (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:42976)
    at o.fire (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:43737)
    at Y (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:30621)
    at r.invoke (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:42976)
    at o.fire (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:43737)
    at d.fireClose (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:51618)
    at ChildProcess.<anonymous> (/Users/kongweiying/.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/client/bundle.js:1:62579)
    at ChildProcess.emit (events.js:327:22)
    at maybeClose (internal/child_process.js:1048:16)
    at Socket.<anonymous> (internal/child_process.js:439:11)
    at Socket.emit (events.js:315:20)
    at Pipe.<anonymous> (net.js:673:12)
[2022-03-19 13:48:39.063] [exthost] [info] ExtensionService#_doActivateExtension zenclabs.previewjs, startup: false, activationEvent: 'onLanguage:typescriptreact'
[2022-03-19 13:48:39.080] [exthost] [error] Activating extension zenclabs.previewjs failed due to an error:
[2022-03-19 13:48:39.080] [exthost] [error] Error: Command failed with ENOENT: npm install
spawn npm ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:269:19)
    at onErrorNT (internal/child_process.js:465:16)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
```

**Comment by fwouts** [comment]
If you've just installed npm, it's possible you need to restart VS Code first. Could you try that?

**Comment by fwouts** [comment]
Can you also confirm that npm is in your PATH?

**Comment by TheAussieStew** [comment]
I think I have it in my PATH:

In my .zshrc I have:

```
eval "$(/opt/homebrew/bin/npm shellenv)"
```

I didn't have that in my .zprofile but I added it anyway and tried reinstalling, however it didn't work. I mean if the npm command works in the shell, then does that mean npm is properly installed in PATH?

**Comment by TheAussieStew** [comment]
Here's what comes up in the Log (Extension Host):

```
[2022-03-19 15:54:38.101] [exthost] [info] Extension host with pid 46553 started
[2022-03-19 15:54:38.101] [exthost] [info] Skipping acquiring lock for /Users/kongweiying/Library/Application Support/Code/User/workspaceStorage/1f9645489d518dd927362d967cebcd0f.
[2022-03-19 15:54:38.130] [exthost] [info] ExtensionService#_doActivateExtension vscode.microsoft-authentication, startup: false, activationEvent: 'onAuthenticationRequest:microsoft'
[2022-03-19 15:54:38.136] [exthost] [info] ExtensionService#_doActivateExtension vscode.debug-auto-launch, startup: true, activationEvent: '*'
[2022-03-19 15:54:38.137] [exthost] [info] ExtensionService#_doActivateExtension vscode.git-base, startup: true, activationEvent: '*', root cause: vscode.github
[2022-03-19 15:54:38.138] [exthost] [info] ExtensionService#_doActivateExtension vscode.ipynb, startup: true, activationEvent: '*'
[2022-03-19 15:54:38.139] [exthost] [info] ExtensionService#_doActivateExtension vscodevim.vim, startup: true, activationEvent: '*'
[2022-03-19 15:54:38.161] [exthost] [info] ExtensionService#_doActivateExtension vscode.npm, startup: true, activationEvent: 'workspaceContains:package.json'
[2022-03-19 15:54:38.282] [exthost] [info] ExtensionService#_doActivateExtension vscode.git, startup: true, activationEvent: '*'
[2022-03-19 15:54:38.290] [exthost] [info] ExtensionService#_doActivateExtension vscode.github, startup: true, activationEvent: '*'
[2022-03-19 15:54:38.319] [exthost] [info] ExtensionService#_doActivateExtension vscode.github-authentication, startup: false, activationEvent: 'onAuthenticationRequest:github'
[2022-03-19 15:54:38.377] [exthost] [info] Eager extensions activated
[2022-03-19 15:54:38.408] [exthost] [info] ExtensionService#_doActivateExtension vscode.emmet, startup: false, activationEvent: 'onStartupFinished'
[2022-03-19 15:54:38.411] [exthost] [info] ExtensionService#_doActivateExtension vscode.merge-conflict, startup: false, activationEvent: 'onStartupFinished'
[2022-03-19 15:54:38.412] [exthost] [info] ExtensionService#_doActivateExtension zenclabs.previewjs, startup: false, activationEvent: 'onStartupFinished'
[2022-03-19 15:54:38.431] [exthost] [error] Activating extension zenclabs.previewjs failed due to an error:
[2022-03-19 15:54:38.431] [exthost] [error] Error: Cannot find module '@previewjs/core'
Require stack:
- /Users/kongweiying/.vscode/extensions/zenclabs.previewjs-1.4.1/dist/index.js
- /private/var/folders/hs/dkcc1ntx5fz50h9phfbv_8900000gn/T/AppTranslocation/92464631-A710-4B66-BF2A-697E0812F4C9/d/Visual Studio Code.app/Contents/Resources/app/out/vs/loader.js
- /private/var/folders/hs/dkcc1ntx5fz50h9phfbv_8900000gn/T/AppTranslocation/92464631-A710-4B66-BF2A-697E0812F4C9/d/Visual Studio Code.app/Contents/Resources/app/out/bootstrap-amd.js
- /private/var/folders/hs/dkcc1ntx5fz50h9phfbv_8900000gn/T/AppTranslocation/92464631-A710-4B66-BF2A-697E0812F4C9/d/Visual Studio Code.app/Contents/Resources/app/out/bootstrap-fork.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:934:15)
    at Function.h.resolve (/private/var/folders/hs/dkcc1ntx5fz50h9phfbv_8900000gn/T/AppTranslocation/92464631-A710-4B66-BF2A-697E0812F4C9/d/Visual Studio Code.app/Contents/Resources/app/out/vs/loader.js:4:761)
    at requireModule (/Users/kongweiying/.vscode/extensions/zenclabs.previewjs-1.4.1/dist/index.js:7572:34)
    at load2 (/Users/kongweiying/.vscode/extensions/zenclabs.previewjs-1.4.1/dist/index.js:7567:20)
    at activate (/Users/kongweiying/.vscode/extensions/zenclabs.previewjs-1.4.1/dist/index.js:7848:50)
    at async Promise.all (index 0)
[2022-03-19 15:54:46.464] [exthost] [info] ExtensionService#_doActivateExtension vscode.typescript-language-features, startup: false, activationEvent: 'onLanguage:typescriptreact'
```

**Comment by fwouts** [comment]
This should be fixed by 8e9a36d :)
